Tests/SonarScanner.MSBuild.PreProcessor.Test/JreCaching/JreCacheTests.cs

```diff
@@ -78,8 +78,9 @@ public void CacheHomeCreationFails(Type exceptionType)
         var directoryWrapper = Substitute.For<IDirectoryWrapper>();
         directoryWrapper.When(x => x.CreateDirectory(cache)).Throw((Exception)Activator.CreateInstance(exceptionType));
         var fileWrapper = Substitute.For<IFileWrapper>();
+        var checksum = Substitute.For<IChecksum>();
 
-        var sut = new JreCache(testLogger, directoryWrapper, fileWrapper);
```

Comment:
I am leaving this here, but this applies to the entire file, and it is a follow up of the previous PR.
There is a lot of duplication on this file, I suggest this approach at the top of the file, and you can set any method-specific logic on the substitutes:

```csharp
    private readonly TestLogger testLogger;
    private readonly IDirectoryWrapper directoryWrapper;
    private readonly IFileWrapper fileWrapper;
    private IChecksum checksum;
    private IJreCache sut;

    public JreCacheTests()
    {
        testLogger = new TestLogger();
        directoryWrapper = Substitute.For<IDirectoryWrapper>();
        fileWrapper = Substitute.For<IFileWrapper>();
        checksum = Substitute.For<IChecksum>();
    }

    [TestInitialize]
    public void Initialize()
    {
        sut = new JreCache(testLogger, directoryWrapper, fileWrapper, checksum);
    }
```

Reply:
Good idea. I will come back to it at a later point.

Reply:
I would urge you to do it here, as it will be very helpful with reviewing the tests, both on this and on follow-up PRs.